python/shotgun_globals/cached_schema.py

```diff
@@ -735,43 +746,63 @@ def get_status_color(cls, status_code, project_id=None):
         return status_color
 
     @classmethod
-    def field_is_editable(cls, sg_entity_type, field_name):
+    def field_is_editable(cls, sg_entity_type, field_name, project_id=None):
         """
         Returns a boolean identifying the editability of the entity's field.
 
         :param str sg_entity_type: the entity type
         :param str field_name: the field name to check editibility
+        :param project_id:  The id of the project.
+                            If None, the current context's project will be used.
 
         :returns: ``True`` if the field is ediable, ``False`` otherwise.
         """
         self = cls.__get_instance()
-        self._check_schema_refresh(sg_entity_type, field_name)
 
-        if sg_entity_type in self._type_schema and field_name in self._field_schema[sg_entity_type]:
-            data = self._field_schema[sg_entity_type][field_name]
+        project_id = project_id or self._get_current_project_id()
+        self._check_schema_refresh(sg_entity_type, field_name, project_id=project_id)
+
+        # make sure the project id is found in each of the type and file schemas
+        # and that the entity type and field name are found in their respective
+        # project caches
+        if (project_id in self._type_schema and
+            project_id in self._field_schema and
+            sg_entity_type in self._type_schema[project_id] and
+            field_name in self._field_schema[project_id][sg_entity_type]):
+
+            data = self._field_schema[project_id][sg_entity_type][field_name]
             try:
                 return data["editable"]["value"]
             except KeyError:
                 raise ValueError("Could not determine editability from the schema.")
 
-
         raise ValueError("Could not find the schema for %s.%s" % (sg_entity_type, field_name))
 
     @classmethod
-    def field_is_visible(cls, sg_entity_type, field_name):
+    def field_is_visible(cls, sg_entity_type, field_name, project_id=None):
         """
         Returns a boolean identifying the visibility of the entity's field.
 
         :param sg_entity_type: the entity type
         :param field_name: the field name to check visibility
+        :param project_id:  The id of the project.
+                            If None, the current context's project will be used.
```

Comment:
Yes, I added this while that work was going on. When I merge this into master, i'll make sure to keep those changes and not these.